physical/raft/raft_autopilot.go

```diff
@@ -425,6 +426,13 @@ func (d *Delegate) KnownServers() map[raft.ServerID]*autopilot.Server {
 			Ext:         d.autopilotServerExt(state),
 		}
 
+		// Use existing NodeType from autopilot state if present, or fallback to a more basic check.
```

Comment:
I feel like we can make this comment more helpful, rather than just being an English version of the code.  Some questions I have that would be nice to cover:
- why is the fallback needed, under what circumstances?
- what if neither branch is taken?
- why do we need to ask autopilot for information to provide to autopilot?